basis/windows/dropfiles/dropfiles.factor

```diff
@@ -0,0 +1,63 @@
+! Copyright (C) 2017-2018 Alexander Ilin.
+! See http://factorcode.org/license.txt for BSD license.
+USING: accessors alien alien.data alien.libraries alien.strings
+continuations fry io.encodings.utf16n kernel literals math
+namespaces sequences ui.backend.windows ui.gadgets.worlds
+ui.gestures windows.messages windows.shell32 windows.types
+windows.user32 ;
+IN: windows.dropfiles
+
+: filecount-from-hdrop ( hdrop -- n )
+    0xFFFFFFFF f 0 DragQueryFile ;
+
+: filenames-from-hdrop ( hdrop -- filenames )
+    dup filecount-from-hdrop <iota>
+    [
+        2dup f 0 DragQueryFile 1 + ! get size of filename buffer
+        dup WCHAR <c-array>
+        [ swap DragQueryFile drop ] keep
+        utf16n alien>string
+    ] with map ;
+
+! : point-from-hdrop ( hdrop -- loc )
```

Comment:
This will be useful if we want to find out the exact mouse coordinates of the drop location within a `world`. Currently I rely on `hand-gadget` to be updated with the drop target gadget, but it is feasible to have gadgets that handle drops differently depending on the location. In any case, it presumes a future protocol extension, which isn't there at the moment. Maybe active mouse-with-drop tracking, visual feedback about actions to be performed on drop, etc.

In short, this is some that may be useful in the future.